addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XXEPlugin.java

```diff
@@ -293,19 +299,13 @@ public void scan() {
 
             try {
                 String requestBody = msg.getRequestBody().toString();
-                Matcher matcher = tagPattern.matcher(requestBody);
+                Matcher matcher;
                 StringBuilder sb = new StringBuilder(ATTACK_HEADER);
                 String localFile;
                 String response;
-                int endIdx = 0;
-
-                while (matcher.find()) {
-                    sb.append(requestBody.substring(endIdx, matcher.start(1)));
-                    sb.append(ATTACK_ENTITY);
-                    endIdx = matcher.end(1);
-                }
-
-                sb.append(requestBody.substring(endIdx));
+                String modifiedBody =
+                        replaceElementValuesWithAttackEntityAndRemoveHeader(requestBody);
+                sb.append(modifiedBody);
                 requestBody = sb.toString();
 
                 for (int idx = 0; idx < LOCAL_FILE_TARGETS.length; idx++) {
```

Comment:
Isn’t the array index issue here? Shouldn’t this be `<=` vs changing the array content?

Reply:
No, because for every LOCAL_FILE_TARGETS exists a corresponding LOCAL_FILE_PATTERNS that is used for detecting, if there is a hit

Reply:
Sorry. I mixed up the constant names 🙁